src/Util.elm

```diff
@@ -14,17 +13,29 @@ acceptableAccounts query accounts =
         lowerQuery =
             String.toLower query
     in
-        if query == "" then
-            []
-        else
-            List.filter (String.contains lowerQuery << String.toLower << .username) accounts
+    if query == "" then
+        []
 
+    else
+        List.filter (String.contains lowerQuery << String.toLower << .username) accounts
 
-extractAuthCode : Navigation.Location -> Maybe String
-extractAuthCode { search } =
-    case (String.split "?code=" search) of
-        [ _, authCode ] ->
-            Just authCode
+
+
+{-
+   TODO: refactor this code smell
+-}
+
+
+extractAuthCode : Url.Url -> Maybe String
```

Comment:
We might want to use Url.Parser.Query there.

Reply:
Fixed using QS package. `Url.Parser.Query` is more useful for parsing full URLs